analysis/subset_data/SubsetBam.wdl

```diff
@@ -0,0 +1,169 @@
+workflow SubsetBam {
+  # Bam to subset
+  File aligned_gold_bam
+
+  # How many cells to grab from the largest and smallest end
+  Int num_large_cells
+  Int num_small_cells
+
+  # How many reads a cell must have before it can be selected
+  Int? min_reads_per_cell
+
+  # Array of comma separated tag values to put in each fastq e.g ["ORIG", "SR", "CB,UB"] "ORIG" means to
+  # grab the original sequence/qualities from the read
+  Array[String] barcode_tags_to_split_by
+  Array[String] quality_tags_to_split_by
+  Array[String]? separators_to_split_by
+
+  Array[String]? final_fastq_basenames
```

Comment:
Essentially I was trying to give the user the ability to name their final files.  I could be wrong here but the aligned bam created from the 3` pipeline will not be named in any way like the input fastq, I'm thinking like the "I1", "R1", "R2" fastq structure.  The split tool currently does have a way to name the files that it's splitting into(it just puts a "_{index}" at the end of the basename variable passed into the tool.  Since the user would be in control of the order of how the split bams/fastqs are output, they would be able to name their final outputs in what ever convention they seem necessary.

I dont think there is any information from the input_bam that would allow a tool to programmatically know what to call the split bams/fastqs but I would love to talk more about this problem as I dont like the current solution either